plugins/status/plugin.go

```diff
@@ -222,23 +252,30 @@ func (p *Plugin) loop() {
 	ctx, cancel := context.WithCancel(context.Background())
 
 	for {
+
 		select {
 		case statuses := <-p.pluginStatusCh:
 			p.lastPluginStatuses = statuses
```

Comment:
How will multiple plugins report their status at the same time?  Won't this always just save the last status reported, and the others will get dropped without a trigger to flush them?  I had to use a buffered channel to get around this I think.

Reply:
When the plugins (eg, bundle, decision log) update their status, they go through the manager. The manager then makes a copy of the statuses of all plugins and calls the listener function. So the `p.lastPluginStatuses` map should contain the status of all plugins.

Reply:
ah gotcha